src/Table/TableController.h

```diff
@@ -32,7 +33,7 @@ struct TableViewCache {
 
 class TableController {
 public:
-    TableController(const std::string& top_level_folder, const std::string& starting_folder);
+    TableController(ProgramSettings& settings);
```

Comment:
You should not be passing this in as a parameter; you should be setting it in the initializer list, the same way you're doing it in `Session`.